akd/src/tree_node.rs

```diff
@@ -89,6 +224,56 @@ pub struct TreeNode {
     pub hash: [u8; 32],
 }
 
+impl TreeNode {
+    // Storage operations
+    pub(crate) async fn write_to_storage<S: Storage + Send + Sync>(
+        &self,
+        storage: &S,
+    ) -> Result<(), StorageError> {
+        // retrieve the higest node properties, at a previous epoch than this one. If we're modifying "this" epoch, simply take it as no need for a rotation.
+        // When we write the node, with an updated epoch value, we'll rotate the stored value and capture the previous
+        let target_epoch = match self.last_epoch {
+            e if e > 0 => e - 1,
+            other => other,
+        };
+        let previous = match TreeNodeWithPreviousValue::get_from_storage(
```

Comment:
We want to retrieve the previous ```latest_node``` value, so we want to investigate where epoch - 1. When a request comes in to write the node with a future epoch, epoch - 1 will be the latest node in storage and we'll do a shift-left. 

The get call should ideally utilize a cached value, so this should be safe to call repeatedly.